osquery/events/linux/auditdnetlink.cpp

```diff
@@ -460,22 +376,17 @@ bool AuditdNetlink::processThread() noexcept {
       audit_event_record_queue.push_back(audit_event_record);
     }
 
-    // Dispatch the new records to the subscribers
+    // Save the new records and notify the reader
     if (!audit_event_record_queue.empty()) {
-      std::lock_guard<std::mutex> subscriber_list_lock(subscribers_mutex_);
-
-      for (auto& subscriber_descriptor : subscribers_) {
-        auto& subscriber_context = subscriber_descriptor.second;
-
-        std::lock_guard<std::mutex> queue_lock(subscriber_context.queue_mutex);
+      std::lock_guard<std::mutex> queue_lock(event_queue_mutex_);
```

Comment:
See about about our `WriteLock` and `ReadLock` tiny abstractions in `osquery/core.h`.

Reply:
It doesn't look like I can use them with std::condition_variables (which are already timed)